python/lance/__init__.py

```diff
@@ -19,20 +19,19 @@
 import pyarrow as pa
 import pyarrow.compute as pc
 import pyarrow.dataset as ds
-
 from lance.lib import BuildScanner, LanceFileFormat, WriteTable
 from lance.types import register_extension_types
 
 if platform.system() == "Linux":
     # TODO enable on MacOS
     register_extension_types()
 
-__all__ = ["dataset", "write_table", "scanner"]
+__all__ = ["dataset", "write_table", "scanner", "LanceFileFormat"]
 
 
 def dataset(
     uri: str,
-) -> ds.Dataset:
+) -> ds.FileSystemDataset:
```

Comment:
why not leave it more generic? do we use FileSystemDataset-specific APIs?

Reply:
FileSystemDataset has this `dataset.files` attributes.